include/boost/geometry/algorithms/detail/check_iterator_range.hpp

```diff
@@ -59,10 +59,74 @@ struct check_iterator_range
                 return false;
             }
         }
+
         return AllowEmptyRange || first != beyond;
     }
 };
 
+// As check_iterator_range but the index of the element that
+// satisfies the predicate is returned otherwise -1 is returned
+template <typename Predicate>
+struct check_iterator_range_with_index
+{
+
+    // version where we can pass a predicate object
+    template <typename InputIterator>
+    static inline int apply(InputIterator first,
```

Comment:
The return type should be: `std::iterator_traits<InputIterator>::difference_type`

Reply:
indeed, changed.